File: C++/027.cpp
// 027 Remove Element
//
// 移除数组中重复的元素后，数组的长度。即给定要移除的数字，输出结果。
//

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int left = 0;
        for (auto &num : nums)
            if (num != val) nums[left++] = num;
        return left;
    }
};

File: Go/027.go
func removeElement(nums []int, value int) int {
	for i := 0; i < len(nums); {
		if (nums[i] == value) {
			// 将后面的值覆盖前面的值
			nums = append(nums[:i], nums[i+1:]...)
		} else {
			i++
		}
	}
	return len(nums)
}
